### PR TITLE
Improves prompting for RAG, enabling better handling on messages that we have no context and adds plugin system

### PR DESCRIPTION
**Pull request by vmesel**
Changes the way we process context  in order to make it usable

### CODE REVIEW

File: src/llm/__init__.py [hunk 1]
```diff
@@ -94,11 +94,11 @@ async def process_user_intent(session_id, message):
     if len(relevant_contents) > 0:
         prompt_templating.append(
             SystemMessagePromptTemplate.from_template(
-                f"{PROMPT.get('suggested')}\n\n{suggested_content}"
+                f"{PROMPT.get('suggested')}. {suggested_content}"
             )
         )
 
-    prompt_templating.append(HumanMessagePromptTemplate.from_template("{user_message}"))
+    prompt_templating.append(HumanMessagePromptTemplate.from_template("Responda a seguinte pergunta:\n{user_message}"))
```
Comment (avelino) [inline src/llm/__init__.py:101]: if the prompt is in English?
we can't assume that all users are using the knowledge base in Portuguese

**Review by avelino — CHANGES_REQUESTED** [review]

File: src/llm/__init__.py (hunk 1 above)
Comment (vmesel) [reply src/llm/__init__.py:101]: @avelino that's true, my bad!

**Comment by vmesel** [comment]
Added plugin systems from #77. Closes #77.

File: src/main.py [hunk 2]
```diff
@@ -91,4 +91,17 @@ async def get_chat_content(chat_id):
 
 @app.post("/session")
 async def create_session():
-    return db_create_session()
\ No newline at end of file
+    return db_create_session()
+
+
+for plugin in PLUGINS:
```
Comment (avelino) [inline src/main.py:97]: @walison17 review pls

**Comment by avelino** [comment]
@vmesel have you seen this?

https://github.com/madkote/fastapi-plugins

**Comment by vmesel** [comment]
@avelino, nope, I didnt, but I found it too verbose. The purpose here was to be a simple API that would expose other endpoints, we can extend it using this tool, though!

**Comment by vmesel** [comment]
Closing this in favor of #80

**Pull request closed without merging**